### PR TITLE
Adi/feedback submission dao

### PR DESCRIPTION
**Pull request by AdiFinkelman**

### CODE REVIEW

File: src/main/kotlin/com/hibob/project/dao/feedbackConfigurationDao.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class FeedbackDao @Autowired constructor(private val sql: DSLContext) {
+
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/dao/feedbackConfigurationDao.kt:11]: remove blank line
no need for blank lines at the top of the class

File: src/main/kotlin/com/hibob/project/dao/feedbackConfigurationDao.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private val feedbackTable = FeedbackConfigurationTable.instance
+
+    private val feedbackConfigurationMapper = RecordMapper<Record, FeedbackConfiguration>
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/dao/feedbackConfigurationDao.kt:14]: the standard in kotlin is this:
private val feedbackConfigurationMapper = RecordMapper<Record, FeedbackConfiguration> { record ->
...
}

File: src/main/kotlin/com/hibob/project/dao/models.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    val status: StatusType
+)
+
+data class FeedbackCreationRequest(
+    val text: String,
+    val creationTime: Timestamp,
+    val isAnonymous: Boolean,
+    val status: StatusType
+) {
+    fun toFeedbackConfiguration(id: Long, employeeId: Long, companyId: Long): FeedbackConfiguration {
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/dao/models.kt:21]: If this is only for tests, move it to the tests file. There shouldn't be code for tests in the code itself.

If you are planning to use it in general, then my comment is that I don't understand why you need this conversion method

File: src/main/kotlin/com/hibob/project/dao/models.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+data class Feedback(
+    val feedbackConfiguration: FeedbackConfiguration,
+    val commentList: List<Comment>
+)
+
+enum class StatusType {
+    REVIEWED, UNREVIEWED
+}
+
+enum class Role() {
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/dao/models.kt:52]: why the constructor brackets here?

File: src/main/resources/db/migration/V202409231531__create_employee_and_company.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
+-- Create company table
+CREATE TABLE company (
```
Comment (vasofi) [inline src/main/resources/db/migration/V202409231531__create_employee_and_company.sql:2]: In your case, this is not a problem that you have migrations and code in the same PR.
But for general knowledge, in bob the pipeline prevents it. We don't want to mix DDLs and code. So in that case you would have had to fix this issue with the rebase

File: src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Test
+    fun `submit anonymous feedback successfully`() {
+        val feedbackCreation = FeedbackCreationRequest("Anonymous", Timestamp.valueOf(LocalDateTime.now()), true, StatusType.UNREVIEWED)
+        feedbackDao.feedbackSubmission(feedbackCreation, userTest)
+        assertTrue(feedbackDao.getAllFeedbacks(companyIdTest)[0].isAnonymous)
+    }
+
+    @Test
+    fun `get all feedbacks where list is empty and throws exception`() {
```
Comment (vasofi) [inline src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt:47]: the title of the test doesn't fit anymore

File: src/main/kotlin/com/hibob/project/dao/models.kt (hunk 3 above)
Comment (AdiFinkelman) [reply src/main/kotlin/com/hibob/project/dao/models.kt:21]: Got it, Ill move it to test

File: src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            companyId = companyId,
+            text = feedbackCreationRequest.text,
+            creationTime = feedbackCreationRequest.creationTime,
+            isAnonymous = feedbackCreationRequest.isAnonymous,
+            status = feedbackCreationRequest.status
+        )
+        return feedbackConfiguration
+    }
+
+    @BeforeEach
```
Comment (vasofi) [inline src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt:66]: don't use both BeforeEach and AfterEach on the same method. It runs it twice.

(I know you were taught to do that, but it makes it run twice)

File: src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        val feedbackConfiguration = FeedbackConfiguration(
+            id = id,
+            employeeId = employeeId,
+            companyId = companyId,
+            text = feedbackCreationRequest.text,
+            creationTime = feedbackCreationRequest.creationTime,
+            isAnonymous = feedbackCreationRequest.isAnonymous,
+            status = feedbackCreationRequest.status
+        )
+        return feedbackConfiguration
```
Comment (vasofi) [inline src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt:63]: new line before return

File: src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import java.sql.Timestamp
+import java.time.LocalDateTime
+
+@BobDbTest
+class FeedbackDaoTest @Autowired constructor(private val sql: DSLContext) {
+
```
Comment (vasofi) [inline src/test/kotlin/com/hibob/project/dao/FeedbackDaoTest.kt:15]: no new line at the top of class

File: src/main/kotlin/com/hibob/project/dao/feedbackConfigurationDao.kt [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+            companyId = record[feedbackTable.companyId],
+            text = record[feedbackTable.text],
+            creationTime = record[feedbackTable.creationTime],
+            isAnonymous = record[feedbackTable.isAnonymous],
+            status = enumValueOf(record[feedbackTable.status])
+        )
+    }
+
+    fun getAllFeedbacks(companyId: Long): List<FeedbackConfiguration> =
+        sql.select(feedbackTable.id, feedbackTable.employeeId, feedbackTable.companyId, feedbackTable.text, feedbackTable.creationTime, feedbackTable.isAnonymous, feedbackTable.status)
```
Comment (vasofi) [inline src/main/kotlin/com/hibob/project/dao/feedbackConfigurationDao.kt:26]: break the params list to multiple lines

**Review by vasofi — APPROVED** [review]

**Pull request merged**